content/blog/2020-06-28-shtab-completion.md

```diff
@@ -0,0 +1,204 @@
+---
+title: '(Tab) Complete Any Python Application in 1 Minute or Less'
+date: 2020-06-28
+description: |
+  We've made a painless tab-completion script generator for Python applications!
+  Find out how to take advantage of it in this blog post.
+
+descriptionLong: |
+  We've made a painless tab-completion script generator for Python applications!
+  It's called `shtab` and it currently works with `argparse`, `docopt`, and
+  `argopt` to produce `bash` and `zsh` completion scripts.
+  This tool was originally created to help `dvc`, but we realised it could be made
+  more generic and valuable to the world's entire ecosystem of Python CLI
+  applications. Find out how to take advantage of it in this blog post.
+picture: 2020-06-28/tab-py.jpg
+pictureComment: Zero Effort Tab Completion for Python Applications
+author: casper_dcl
+---
+
+Command line tools are powerful. Things like [`make`] have manual pages
+spanning, well,
+[pages](https://www.gnu.org/software/make/manual/make.html#Options-Summary),
+while just the list of [`git`] subcommands is longer than can fit on a standard
+`80 x 24` terminal screen.
+
+```dvc
+$ git <TAB>
+add                  filter-branch        rebase
+am                   format-patch         reflog
+annotate             fsck                 relink
+...
+describe             prco                 unassume
+--More--
+```
+
+Notice the `--More--` at the bottom? That's the joy of pagination.
+
+Notice the `<TAB>` at the top? That represents actually pressing the tab key.
+Ah, the joy of shell tab completion.
+
+Tab completion is an indispensable part of writing anything on the command-line.
+Personally, I can't imaging trying to `git co` (aliased to `git checkout`) a
+branch without `<TAB>` to do the heavy lifting.
+[They say](https://en.wikipedia.org/wiki/Letter_frequency) "E" is the most
+common vowel, and "T" the most common consonant. My keyboard use probably looks
+more like this:
+
+![](/uploads/images/2020-06-28/key-frequencies.png 'Yes, I use vim =500')_My
+key usage
```

Comment:
https://dvc.org/doc/user-guide/contributing/blog implies that ![]()_caption should render as a caption.

This doesn't seem to work for blog posts. CC @rogermparent / @shcheklein

Reply:
nvm missing underscore suffix. `![]()_caption_`